### DISCUSSION TITLE
Bundle size for production build with imported icon font

### DISCUSSION THREAD

**Discussion by davideangelillo00** [lit/lit#4741] [Q&A]
Hi,
I have a web component library built with Lit + Vite and I'm having some issues with the production bundle size as soon as I try to include an icon font inside my CSS.
I'm currently using Phosphor Icon as an asset to reduce this issue, having it imported as a CSS inside my components this way:
![image](https://github.com/user-attachments/assets/6089a4b6-35c4-43b7-840e-da4ad1fc5758)

```ts
import icon from '../../../../public/icons/style.css?inline';
@customElement('my-component')
export class MyComponent extends LitElement {
  // ... my stuff
  public static styles: CSSResultGroup = [
    // ... my other styles
    unsafeCSS(icon),
  ];
}
```
Then the embedding app should include the icon font CSS (exported as an asset inside my component library) inside its index.html.

This way I can get my bundle to have small components (about 10kB) but I also get one huge style-{hash}.js of 10MB.
Trying to import the icon font in the index.scss makes every component weight about 5MB, so it cannot be a solution.

What am I missing? Is there a better way to do it? Is it ok to have a ~10MB component library just because of assets?

**Comment by justinfagnani** [comment]
What tool are you using?

Importing CSS like that is non-standard. The standard syntax is:

```ts
import icon from '../../../../public/icons/style.css?inline' with {type: 'css'};
@customElement('my-component')
export class MyComponent extends LitElement {
  // ... my stuff
  public static styles: CSSResultGroup = [
    // ... my other styles
    icon,
  ];
}
```

Though this isn't supported by all browsers yet, and I'm unsure of the tooling support.

Can you just not bundle your font? Would that event help? If your font is 5MB and you depend on it, that I can't see a way around having smaller assets. Maybe you can make your users bring the font however they wish to?

**Reply by davideangelillo00** [reply]
Hi @justinfagnani ,
syntax matter apart (which I corrected, thanks), the main issue is that even if you import the css of the icon font only outside of the component like in index.html (making users bring the font) would not work on web component due to shadow dom.

Considering the size of Lit web components, I'm assuming that carrying 5/10MB of assets is not a good practice.
Is this assumption correct? Do you think there could be a way to work around this problem?

**Reply by davideangelillo00** [reply]
About the tools, this is my vite.config.ts (both index.ts and index.scss are inside entry array, with other components).

![image](https://github.com/user-attachments/assets/2645206c-0408-48a2-9c94-bc56427e0c06)
